deployment/setup_20.04.sh

```diff
@@ -1,164 +1,282 @@
-echo "To configure a domain and SSL certificate, follow the guide at https://www.chatwoot.com/docs/deployment/deploy-chatwoot-in-linux-vm"
-echo "***************************************************************************"
-else
-curl https://ssl-config.mozilla.org/ffdhe4096.txt >> /etc/ssl/dhparam
-wget https://raw.githubusercontent.com/chatwoot/chatwoot/develop/deployment/nginx_chatwoot.conf
-cp nginx_chatwoot.conf /etc/nginx/sites-available/nginx_chatwoot.conf
-certbot certonly --non-interactive --agree-tos --nginx -m $le_email -d $domain_name
-sed -i "s/chatwoot.domain.com/$domain_name/g" /etc/nginx/sites-available/nginx_chatwoot.conf
-ln -s /etc/nginx/sites-available/nginx_chatwoot.conf /etc/nginx/sites-enabled/nginx_chatwoot.conf
-systemctl restart nginx
-sudo -i -u chatwoot << EOF
-cd chatwoot
-sed -i "s/http:\/\/0.0.0.0:3000/https:\/\/$domain_name/g" .env
+  systemctl enable chatwoot.target
+  systemctl start chatwoot.target
+}
+
+
+function setup_ssl() {
+  echo "debug: setting up ssl"
+  echo "debug: domain: $domain_name"
+  echo "debug: letsencrypt email: $le_email"
+  curl https://ssl-config.mozilla.org/ffdhe4096.txt >> /etc/ssl/dhparam
+  wget https://raw.githubusercontent.com/chatwoot/chatwoot/develop/deployment/nginx_chatwoot.conf
+  cp nginx_chatwoot.conf /etc/nginx/sites-available/nginx_chatwoot.conf
+  certbot certonly --non-interactive --agree-tos --nginx -m $le_email -d $domain_name
+  sed -i "s/chatwoot.domain.com/$domain_name/g" /etc/nginx/sites-available/nginx_chatwoot.conf
+  ln -s /etc/nginx/sites-available/nginx_chatwoot.conf /etc/nginx/sites-enabled/nginx_chatwoot.conf
+  systemctl restart nginx
+  sudo -i -u chatwoot << EOF
+  cd chatwoot
+  sed -i "s/http:\/\/0.0.0.0:3000/https:\/\/$domain_name/g" .env
 EOF
-systemctl restart chatwoot.target
-echo -en "\n\n***************************************************************************\n"
-echo "Woot! Woot!! Chatwoot server installation is complete"
-echo "The server will be accessible at https://$domain_name"
-echo "***************************************************************************"
-fi
+  systemctl restart chatwoot.target
+}
 
+function setup_logging() {
+  touch /var/log/chatwoot-setup.log
+  LOG_FILE="/var/log/chatwoot-setup.log"
+}
 
-if [ $install_pg_redis == "no" ]
-then
-echo -en "\n\n***************************************************************************\n"
-echo "DB migrations are not run as pg and redis is not installed."
-echo "After modifying .env with your external db creds, run db migrations !!!"
-echo "***************************************************************************"
-fi
+function main() {
+
+  setup_logging
+  cat << EOF
+
+***************************************************************************
+              Chatwoot Installation (latest)
+***************************************************************************
+
+Open up a second terminal and follow along using, tail -f /var/log/chatwoot.
```

Comment:
Since this is the very first sentence output once we start executing the script, it feels that opening a new terminal is required which is not the case. 

Let us add some context before this sentence and place this in an if condition.